main.tf

```diff
@@ -482,7 +477,7 @@ resource "aws_security_group" "fargate_service_sg" {
 resource "aws_ecs_service" "service" {
   name             = local.service_name
   task_definition  = aws_ecs_task_definition.task_def.arn
-  cluster          = local.create_new_cluster ? aws_ecs_cluster.new_cluster[0].id : data.aws_ecs_cluster.existing_cluster[0].id # if cluster is not provided use created one, else use existing cluster
+  cluster          = local.create_new_cluster ? aws_ecs_cluster.new_cluster[0].id : var.existing_ecs_cluster.id # if cluster is not provided use created one, else use existing cluster
```

Comment:
I believe we can use the `.arn` instead of `.id`, so that means we probably don't need the "id" field in the existing_ecs_cluster variable.
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service